### PR TITLE
Log and report unhandled exceptions

### PR DESCRIPTION
**Pull request by pobiega**
This is an "update" on #298

### CODE REVIEW

File: src/Sidekick/App.xaml.cs [hunk 1]
```diff
@@ -49,10 +52,15 @@ protected override async void OnStartup(StartupEventArgs e)
 
             base.OnStartup(e);
 
-            ToolTipService.ShowDurationProperty.OverrideMetadata(
-            typeof(DependencyObject), new FrameworkPropertyMetadata(int.MaxValue));       // Tooltip opened indefinitly until mouse is moved
+            AttachErrorHandlers();
+
+            // Tooltip opened indefinitly until mouse is moved.
```
Comment (domialex) [inline src/Sidekick/App.xaml.cs]: Just noticed this lol: `indefinitely`

File: src/Sidekick/App.xaml.cs [hunk 2]
```diff
@@ -164,5 +172,33 @@ private void EnsureSingleInstance()
...
+            TaskScheduler.UnobservedTaskException += (s, e) =>
+            {
+                LogUnhandledException(e.Exception);
+                e.SetObserved();
+            };
+        }
+
+        private void LogUnhandledException(Exception ex)
+        {
+            logger.Fatal(ex, "Unhandled exception in application root");
```
Comment (domialex) [inline src/Sidekick/App.xaml.cs]: I'm pretty sure we need to try/catch everything here.

File: src/Sidekick/App.xaml.cs [hunk 3]
```diff
@@ -164,5 +172,33 @@ private void EnsureSingleInstance()
...
+            AdonisUI.Controls.MessageBox.Show(ApplicationResources.FatalErrorOccured, buttons: AdonisUI.Controls.MessageBoxButton.OK);
```
Comment (domialex) [inline src/Sidekick/App.xaml.cs]: My first draft was also to shutdown the app, but I think that showing a message box right in front of a player while Sidekick crashes can be very dangerous if he's in hardcore for example.

File: src/Sidekick/App.xaml.cs [hunk 4]
```diff
@@ -164,5 +172,33 @@ private void EnsureSingleInstance()
                 Current.Shutdown();
             }
         }
+
+        private void AttachErrorHandlers()
+        {
+            AppDomain.CurrentDomain.UnhandledException += (s, e) =>
```
Comment (domialex) [inline src/Sidekick/App.xaml.cs]: Just a note, this object has the `IsTerminating` flag, which tells when the exception is actually closing the app.

File: src/Sidekick/App.xaml.cs (hunk 2 above)
Comment (pobiega) [reply src/Sidekick/App.xaml.cs]: Serilog will never throw an uncaught exception when logging, no need to try/catch.

**Pull request merged**